node/router/messages/src/lib.rs

```diff
@@ -120,6 +120,36 @@ impl<N: Network> Message<N> {
         Self::VERSIONS.last().map(|(_, version)| *version).unwrap_or(0)
     }
 
+    /// Returns the lowest acceptable message version for the given block height.
+    pub fn lowest_accepted_message_version(current_block_height: u32) -> u32 {
+        // Fetch the latest message version.
+        let latest_message_version = Self::latest_message_version();
+
+        // Fetch the versions.
+        let versions = Self::VERSIONS;
+
+        // Determine the minimum accepted message version.
+        // Example scenario:
+        // - At block height `X`, the protocol upgrades to message version from `Y-1` to `Y`.
+        // - Client A upgrades and starts using message version `Y`.
+        // - Client B has not upgraded and still uses message version `Y-1`.
+        // - Until block `X`, they stay connected and can communicate.
+        // - After block `X`, Client A will reject messages from Client B.
```

Comment:
Better as a docblock for `lowest_accepted_message_version` I think.

Reply:
Resolved with [80c308a](https://github.com/ProvableHQ/snarkOS/pull/3596/commits/80c308a6cd540d7e5d5f3b9a7daa3ff1add2d9d1)